AssertionError: 1730160000.0 is not an instance of <class 'int'>

### Information about bug

![image](https://github.com/user-attachments/assets/1a2c40b0-ea7d-4df1-86dd-e8d961a4790a)


### Module

stock

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.34.0 (pre-prod)

Frappe Framework: v15.41.0 (pre-prod)


### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell
Traceback (most recent call last):
  File "/home/hrishikesh/8848digital/postgres/frappe-bench-postgres/apps/erpnext/erpnext/stock/doctype/item/test_item.py", line 635, in test_heatmap_data
    self.assertIsInstance(timestamp, int)
AssertionError: 1730160000.0 is not an instance of <class 'int'>
```

--- a/erpnext/stock/doctype/item/item.py
+++ b/erpnext/stock/doctype/item/item.py
@@ -1092,15 +1092,22 @@ def get_timeline_data(doctype: str, name: str) -> dict[int, int]:
     one_year_ago = datetime.now() - timedelta(days=365)
     
     sle = frappe.qb.DocType("Stock Ledger Entry")
+    result = frappe.db.sql(
+		"""
+		SELECT EXTRACT(EPOCH FROM posting_date) AS posting_date_unix,
+			COUNT(*) AS item_count
+		FROM `tabStock Ledger Entry`
+		WHERE item_code = %s
+		AND posting_date > %s
+		GROUP BY posting_date
+		""",
+    (name, one_year_ago),
+    as_dict=True
+	)
 
-    return dict(
-        frappe.qb.from_(sle)
-        .select(UnixTimestamp(sle.posting_date), Count("*"))
-        .where(sle.item_code == name)
-        .where(sle.posting_date > one_year_ago)  # Use the calculated date here
-        .groupby(sle.posting_date)
-        .run()
-    )
+    result_dict = {int(row['posting_date_unix']): row['item_count'] for row in result}
+
+    return result_dict
 
 def validate_end_of_life(item_code, end_of_life=None, disabled=None):
 	if (not end_of_life) or (disabled is None):

